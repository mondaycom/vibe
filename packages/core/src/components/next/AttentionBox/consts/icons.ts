import { Info, Check, Alert, Warning, Favorite } from "@vibe/icons";
import { AttentionBoxType } from "../AttentionBox.types";
import { SubIcon } from "../../../../types";

export const ATTENTION_BOX_DEFAULT_ICONS: Record<AttentionBoxType, SubIcon> = {
  primary: Info,
  success: Check,
  danger: Alert,
  warning: Warning,
  dark: Favorite
};
