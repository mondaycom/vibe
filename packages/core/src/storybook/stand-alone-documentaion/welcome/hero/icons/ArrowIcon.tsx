import React from "react";

export default function ArrowIcon({ className }: { className?: string }) {
  return (
    <svg
      className={className}
      width="146"
      height="89"
      viewBox="0 0 146 89"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M142.191 52.8681L107.627 41.1405M142.191 52.8681L114.507 85.239M142.191 52.8681C50.1209 94.2659 34.4833 12.7837 63.7408 4.73466C92.9982 -3.31434 96.5293 100.303 4.0957 57.5865"
        stroke="#FF5A85"
        strokeWidth="7.14286"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
}
