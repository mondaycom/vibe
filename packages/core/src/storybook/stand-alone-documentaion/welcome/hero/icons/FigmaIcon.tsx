import React from "react";

export default function FigmaIcon() {
  return (
    <svg width="22" height="22" viewBox="0 0 22 22" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M10.9333 11.2388C10.9333 10.3743 11.2767 9.54534 11.888 8.9341C12.4992 8.32288 13.3282 7.97949 14.1926 7.97949C15.057 7.97949 15.886 8.32288 16.4973 8.9341C17.1085 9.54534 17.4519 10.3743 17.4519 11.2388C17.4519 12.1032 17.1085 12.9322 16.4973 13.5434C15.886 14.1546 15.057 14.498 14.1926 14.498C13.3282 14.498 12.4992 14.1546 11.888 13.5434C11.2767 12.9322 10.9333 12.1032 10.9333 11.2388Z"
        fill="#1ABCFE"
      />
      <path
        d="M4.41455 17.7568C4.41455 16.8924 4.75794 16.0634 5.36917 15.4521C5.98039 14.8409 6.8094 14.4976 7.67381 14.4976H10.9331V17.7568C10.9331 18.6212 10.5897 19.4503 9.97846 20.0615C9.36723 20.6726 8.53822 21.0161 7.67381 21.0161C6.8094 21.0161 5.98039 20.6726 5.36917 20.0615C4.75794 19.4503 4.41455 18.6212 4.41455 17.7568Z"
        fill="#0ACF83"
      />
      <path
        d="M10.9333 1.46045V7.97897H14.1926C15.057 7.97897 15.886 7.63558 16.4973 7.02436C17.1085 6.41313 17.4519 5.58412 17.4519 4.71971C17.4519 3.8553 17.1085 3.02629 16.4973 2.41506C15.886 1.80383 15.057 1.46045 14.1926 1.46045H10.9333Z"
        fill="#FF7262"
      />
      <path
        d="M4.41455 4.71971C4.41455 5.58412 4.75794 6.41313 5.36917 7.02436C5.98039 7.63558 6.8094 7.97897 7.67381 7.97897H10.9331V1.46045H7.67381C6.8094 1.46045 5.98039 1.80383 5.36917 2.41506C4.75794 3.02629 4.41455 3.8553 4.41455 4.71971Z"
        fill="#F24E1E"
      />
      <path
        d="M4.41455 11.2388C4.41455 12.1032 4.75794 12.9322 5.36917 13.5434C5.98039 14.1546 6.8094 14.498 7.67381 14.498H10.9331V7.97949H7.67381C6.8094 7.97949 5.98039 8.32288 5.36917 8.9341C4.75794 9.54534 4.41455 10.3743 4.41455 11.2388Z"
        fill="#A259FF"
      />
    </svg>
  );
}
