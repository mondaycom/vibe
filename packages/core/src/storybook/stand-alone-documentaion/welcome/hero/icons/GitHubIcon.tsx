import React from "react";

export default function GitHubIcon() {
  return (
    <svg width="16" height="16" viewBox="0 0 22 22" fill="none" xmlns="http://www.w3.org/2000/svg">
      <g clipPath="url(#clip0_13045_6128)">
        <path
          d="M10.6667 0.831055C4.77333 0.831055 0 5.60705 0 11.4977C0 16.2115 3.056 20.2088 7.29333 21.6177C7.82667 21.7182 8.02222 21.3884 8.02222 21.1048C8.02222 20.8515 8.01333 20.1804 8.00889 19.2915C5.04178 19.9351 4.416 17.8604 4.416 17.8604C3.93067 16.6293 3.22933 16.3004 3.22933 16.3004C2.26311 15.6391 3.304 15.6524 3.304 15.6524C4.37511 15.7271 4.93778 16.7511 4.93778 16.7511C5.88889 18.3822 7.43467 17.9111 8.04445 17.6382C8.14044 16.9484 8.41511 16.4782 8.72 16.2115C6.35111 15.9448 3.86133 15.0275 3.86133 10.9404C3.86133 9.77594 4.27467 8.82483 4.95911 8.07816C4.83911 7.80883 4.47911 6.72439 5.05244 5.25505C5.05244 5.25505 5.94578 4.96883 7.98578 6.34839C8.83911 6.11105 9.74578 5.99372 10.6524 5.98839C11.5591 5.99372 12.4658 6.11105 13.3191 6.34839C15.3458 4.96883 16.2391 5.25505 16.2391 5.25505C16.8124 6.72439 16.4524 7.80883 16.3458 8.07816C17.0258 8.82483 17.4391 9.77594 17.4391 10.9404C17.4391 15.0382 14.9458 15.9404 12.5724 16.2026C12.9458 16.5226 13.2924 17.1768 13.2924 18.1759C13.2924 19.6035 13.2791 20.7502 13.2791 21.0968C13.2791 21.3768 13.4658 21.7102 14.0124 21.6035C18.28 20.2044 21.3333 16.2044 21.3333 11.4977C21.3333 5.60705 16.5573 0.831055 10.6667 0.831055Z"
          fill="black"
        />
      </g>
      <defs>
        <clipPath id="clip0_13045_6128">
          <rect width="21.3333" height="21.3333" fill="white" transform="translate(0 0.571777)" />
        </clipPath>
      </defs>
    </svg>
  );
}
