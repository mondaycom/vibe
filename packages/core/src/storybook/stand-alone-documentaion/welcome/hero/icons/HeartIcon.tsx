import React from "react";

export default function HeartIcon({ className }: { className?: string }) {
  return (
    <svg
      className={className}
      width="114"
      height="113"
      viewBox="0 0 114 113"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M65.0949 41.0114C96.6651 -40.9729 148.872 29.2081 60.0924 109.142C-31.1555 73.6569 4.21914 -45.1488 65.0949 41.0114Z"
        stroke="#FF5A85"
        strokeWidth="7.61905"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
}
