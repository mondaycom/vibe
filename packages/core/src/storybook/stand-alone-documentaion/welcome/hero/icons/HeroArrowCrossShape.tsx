import React from "react";

export default function HeroArrowCrossShape() {
  return (
    <svg width="150" height="150" viewBox="0 0 170 169" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M73.3834 38.241C69.6768 32.347 61.977 30.3634 55.8724 33.7871L55.7273 33.8698C49.5521 37.435 47.3735 45.2618 50.7632 51.4922C50.7796 51.5225 50.7964 51.5527 50.8133 51.5828L50.8834 51.7076L50.9271 51.7843L62.4548 71.751L39.3987 71.7511L39.3079 71.7516L39.1647 71.7534L39.0637 71.7552C32.0288 71.9335 26.3733 77.6446 26.2854 84.6986L26.2843 84.8655L26.2854 85.0324C26.3733 92.0867 32.0292 97.7975 39.064 97.9757L39.1579 97.9774L39.3009 97.9792C39.3335 97.9797 39.3661 97.9799 39.3987 97.9799H62.4532L50.9262 117.945L50.8825 118.022L50.8124 118.147L50.7622 118.237C47.3726 124.468 49.5511 132.294 55.7263 135.86L55.8714 135.942C62.0245 139.393 69.7981 137.351 73.4699 131.347L73.5205 131.263L73.5935 131.14L73.6409 131.06L85.1687 111.092L96.6967 131.06L96.7824 131.206C100.443 137.365 108.388 139.453 114.611 135.86C120.835 132.266 122.999 124.342 119.495 118.092L119.411 117.945L107.884 97.9799H130.938L131.107 97.9789C138.272 97.8881 144.052 92.0519 144.052 84.8656V84.8654C144.052 77.6229 138.181 71.751 130.938 71.751H107.883L119.41 51.7843L119.494 51.6369C122.97 45.4361 120.868 37.5877 114.757 33.9554L114.61 33.8698C108.387 30.2768 100.443 32.3638 96.7815 38.5235L96.6958 38.6698L85.1687 58.6351L73.6418 38.6699L73.5944 38.5889L73.5214 38.466L73.4708 38.3822L73.3834 38.241Z"
        fill="#121212"
      />
      <path
        d="M103.968 42.5827C105.367 40.4206 108.23 39.7133 110.485 41.0153C112.741 42.3173 113.559 45.1508 112.387 47.4431L112.268 47.6628L93.5967 80.0026H130.938L131.188 80.0085C133.759 80.1387 135.804 82.2649 135.805 84.8688C135.805 87.4728 133.76 89.5996 131.189 89.7302L130.938 89.736H93.5997L112.269 122.072L112.389 122.293C113.561 124.585 112.742 127.418 110.487 128.72C108.232 130.022 105.369 129.315 103.97 127.153L103.839 126.939L85.168 94.6003L66.4981 126.938C65.1542 129.266 62.1775 130.064 59.8497 128.72C57.5219 127.376 56.7237 124.4 58.0674 122.072L76.7364 89.736H39.3985C36.7106 89.736 34.5316 87.5577 34.5313 84.8698C34.5313 82.1818 36.7105 80.0027 39.3985 80.0026H76.7403L58.0694 47.6638C56.7255 45.3359 57.5229 42.3594 59.8506 41.0153C62.1785 39.6714 65.155 40.4689 66.4991 42.7966L85.1671 75.1325L103.837 42.7966L103.968 42.5827Z"
        fill="#FFC300"
      />
    </svg>
  );
}
