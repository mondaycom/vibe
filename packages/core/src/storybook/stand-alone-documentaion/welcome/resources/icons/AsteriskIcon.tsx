import React from "react";

export default function AsteriskIcon() {
  return (
    <svg width="49" height="49" viewBox="0 0 49 49" fill="none" xmlns="http://www.w3.org/2000/svg">
      <g id="Resource icons">
        <path
          id="Union"
          d="M16.8267 41.0046C16.516 41.5427 15.828 41.727 15.2899 41.4164C14.7519 41.1057 14.5675 40.4177 14.8782 39.8796L22.9538 25.8921L6.80371 25.8921C6.18239 25.8921 5.67871 25.3885 5.67871 24.7671C5.67871 24.1458 6.18239 23.6421 6.80371 23.6421L22.9554 23.6421L14.8795 9.65439C14.5689 9.11631 14.7532 8.42827 15.2913 8.11761C15.8294 7.80695 16.5174 7.99131 16.8281 8.52939L24.9031 22.5158L32.9787 8.52851C33.2894 7.99043 33.9774 7.80607 34.5155 8.11673C35.0536 8.42739 35.2379 9.11543 34.9273 9.65351L26.8509 23.6421H43.0048C43.6261 23.6421 44.1298 24.1458 44.1298 24.7671C44.1298 25.3885 43.6261 25.8921 43.0048 25.8921L26.8525 25.8921L34.9287 39.8805C35.2393 40.4186 35.055 41.1066 34.5169 41.4173C33.9788 41.7279 33.2908 41.5436 32.9801 41.0055L24.9031 27.0158L16.8267 41.0046Z"
          fill="#323338"
        />
      </g>
    </svg>
  );
}
