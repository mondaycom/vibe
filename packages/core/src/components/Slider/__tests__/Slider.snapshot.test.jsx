import { vi, describe, it, expect } from "vitest";
import React from "react";
import { Sound } from "@vibe/icons";
import { renderSliderForSnapshots } from "./sliderTestUtils.jsx";

vi.mock("../../Icon/Icon", () => ({
  default: ({ icon, ...rest }) => {
    return <div data-testid="mock-icon">{JSON.stringify({ ...rest, icon: icon && icon.displayName })}</div>;
  }
}));

vi.mock("../../TextField/TextField", () => ({
  default: props => {
    return <div data-testid="mock-text-field-comp">{JSON.stringify(props)}</div>;
  }
}));

describe("Slider renders correctly", () => {
  it("01. with empty props", async () => {
    const render = await renderSliderForSnapshots();
    expect(render).toMatchSnapshot();
  });

  it("02. with ariaLabel", async () => {
    const render = await renderSliderForSnapshots({ ariaLabel: "My Slider" });
    expect(render).toMatchSnapshot();
  });

  it("03. with ariaLabelledby", async () => {
    const render = await renderSliderForSnapshots({ ariaLabelledby: "id-of-the-custom-label" });
    expect(render).toMatchSnapshot();
  });

  it("04. with custom className", async () => {
    const render = await renderSliderForSnapshots({ className: "slide-me" });
    expect(render).toMatchSnapshot();
  });

  it("05. with id and data-testid", async () => {
    const render = await renderSliderForSnapshots(
      { id: "my-slider", "data-testid": "my-slider" },
      "my-slider__thumb-0"
    );
    expect(render).toMatchSnapshot();
  });

  it("06. with disabled", async () => {
    const render = await renderSliderForSnapshots({ disabled: true });
    expect(render).toMatchSnapshot();
  });

  it("07. with color", async () => {
    const render = await renderSliderForSnapshots({ color: "positive" });
    expect(render).toMatchSnapshot();
  });

  it(`08. with size`, async () => {
    const render = await renderSliderForSnapshots({ size: "large" });
    expect(render).toMatchSnapshot();
  });

  it(`09. with min, max, and step`, async () => {
    const render = await renderSliderForSnapshots({ min: 10, max: 20, step: 2 });
    expect(render).toMatchSnapshot();
  });

  it(`10. with showValue`, async () => {
    const render = await renderSliderForSnapshots({ showValue: true });
    expect(render).toMatchSnapshot();
  });

  it(`11. with valueFormatter`, async () => {
    const render = await renderSliderForSnapshots({ valueFormatter: value => `--${value}GB--` });
    expect(render).toMatchSnapshot();
  });

  it(`12. with value (controlled component)`, async () => {
    const render = await renderSliderForSnapshots({ value: 12 });
    expect(render).toMatchSnapshot();
  });

  it(`13. with value, valueText (controlled component)`, async () => {
    const render = await renderSliderForSnapshots({ value: 15, valueText: "15kg" });
    expect(render).toMatchSnapshot();
  });

  it(`14. with indicateSelection`, async () => {
    const render = await renderSliderForSnapshots({ indicateSelection: true });
    expect(render).toMatchSnapshot();
  });

  it(`15. with prefix (string) and postfix (Icon)`, async () => {
    const render = await renderSliderForSnapshots({ prefix: "Vol", postfix: { icon: Sound } });
    expect(render).toMatchSnapshot();
  });

  it(`16. with prefix (custom) and postfix (render-props)`, async () => {
    const render = await renderSliderForSnapshots({
      prefix: <div>Custom component</div>,
      postfix: (value, valueText) => `Render Props result: ${valueText} (${value})`
    });
    expect(render).toMatchSnapshot();
  });

  it(`17. with ranged (ranged slider)`, async () => {
    const render = await renderSliderForSnapshots({ ranged: true });
    expect(render).toMatchSnapshot();
  });

  it(`18. ranged slider with options`, async () => {
    const render = await renderSliderForSnapshots({ defaultValue: [20, 30], min: 10, max: 50, ranged: true, step: 2 });
    expect(render).toMatchSnapshot();
  });
});
