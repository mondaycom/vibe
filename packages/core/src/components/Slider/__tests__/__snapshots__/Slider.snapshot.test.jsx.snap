// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Slider renders correctly > 01. with empty props 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 02. with ariaLabel 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-label="My Slider"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 03. with ariaLabelledby 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-labelledby="id-of-the-custom-label"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 04. with custom className 1`] = `
<div
  class="slider positionTop slide-me"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 05. with id and data-testid 1`] = `
<div
  class="slider positionTop"
  data-testid="my-slider"
  id="my-slider"
>
  <div
    class="base small"
    data-testid="my-slider__base"
  >
    <div
      class="rail small"
      data-testid="my-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small notDisabledThumb"
        data-testid="my-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 06. with disabled 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small disabled"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="true"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 07. with color 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small positive"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track positive"
      />
      <div
        class="filledTrack positive"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb positive small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 08. with size 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base large"
    data-testid="monday-slider__base"
  >
    <div
      class="rail large"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb large notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 09. with min, max, and step 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="20"
        aria-valuemin="10"
        aria-valuenow="10"
        aria-valuetext="10%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 10. with showValue 1`] = `
<div
  class="slider valueShown positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      >
        <label
          class="label colorPrimary positionTop"
        >
          0%
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 11. with valueFormatter 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="--0GB--"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 12. with value (controlled component) 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 12%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="12"
        aria-valuetext="12%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 12%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 13. with value, valueText (controlled component) 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 15%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="15"
        aria-valuetext="15kg"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 15%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 14. with indicateSelection 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
  <div
    class="infix postfix"
    style="width: 60px;"
  >
    <div
      data-testid="mock-text-field-comp"
    >
      {"value":"0%","className":"selectionIndicator"}
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 15. with prefix (string) and postfix (Icon) 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="infix prefix txt"
  >
    Vol
  </div>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
  <div
    class="infix postfix"
  >
    <div
      data-testid="mock-icon"
    >
      {"clickable":false,"iconSize":18,"ignoreFocusStyle":true,"icon":"Sound"}
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 16. with prefix (custom) and postfix (render-props) 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="infix prefix"
  >
    <div>
      Custom component
    </div>
  </div>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
  <div
    class="infix postfix"
  >
    Render Props result: 0% (0)
  </div>
</div>
`;

exports[`Slider renders correctly > 17. with ranged (ranged slider) 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 100%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="100"
        aria-valuetext="100%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 100%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly > 18. ranged slider with options 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 25%; width: 25%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="50"
        aria-valuemin="10"
        aria-valuenow="20"
        aria-valuetext="20%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="50"
        aria-valuemin="10"
        aria-valuenow="30"
        aria-valuetext="30%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 50%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;
